### ISSUE TITLE
exclude_controls should remove the entire metabolite from a given experiment

### ISSUE THREAD

**Issue by KujawinskiLaboratory** [joefutrelle/domdb#38]
Right now, config exclude_controls only removes the control sample. What I would like it to do is to remove the entire metabolite from a given experiment. To see this example, you will need the rpom3ext data I just put into the KujawinskiLaboratory/LCdata repository. You will also need the tps6 data which you already have.

Set config:
ppm_diff 1
rt_diff 45
int_over_controls 0
exclude_controls False

run this query:
search 627.1316 235 temp_data_out1.csv

Now change config: 
config exclude_controls True

run the query again. 

When exclude_controls is 'true' it should remove all the answers from the rpom3ext experiment and you should only see the metabolite that matched the tps6 experiment. 

![capture](https://cloud.githubusercontent.com/assets/9040216/6512130/92ea3ab4-c33f-11e4-9203-1f4c50a28197.JPG)

The figure shows the output of the query run with exclude_controls false. If we change that to exclude_controls true, I would like everything in yellow to go away.

**Comment by joefutrelle** [comment]
Got it--should be straightforward. To run the test above, I need the rpom3ext CSV files that work with the latest code, can you upload them to the shared location? Right now it only has neg files for tps4 and tps6.

**Comment by KujawinskiLaboratory** [comment]
Hmmm. I did upload the rpom3ext files in the neg directory. When I go the LCdata, I see six files (Tps4, Tps6, and Rpom3ext).

**Comment by joefutrelle** [comment]
There it is. I thought you had made the decision to use the Google Drive folder instead and so that's where I expected to see it. I will be happy to use either one but I prefer github.

**Comment by joefutrelle** [comment]
Also I need to up my reading comprehension of your earlier post. Apologies,

**Comment by joefutrelle** [comment]
OK, I'm confused now. When I run this query, I get more than 24 rows back, and the tps6 metabolite *is* found in the controls. So shouldn't it also be excluded?

**Comment by joefutrelle** [comment]
The intensities of the tps6 metabolite in the controls are all zero. I think you may be using an older version of my code. I have to include zero intensity samples in the database now or else I cannot compute average intensities correctly, as N will be wrong if I simply ignore zero intensity samples.

I think this just means that the criterion for being present in controls also includes nonzero intensity, rather than just a 1 in the sample control column.

**Comment by KujawinskiLaboratory** [comment]
I downloaded the code yesterday afternoon, so it seems like it should be current. Let me try again from the beginning.

**Comment by joefutrelle** [comment]
No, you've got it right. My mistake again

**Comment by KujawinskiLaboratory** [comment]
I have asked around...we did include zeros when we averaged the samples and the blanks. Here is an example: 
values in samples: 5  & 10 &  0 (average = 5)
values in blanks: 2 & 0 & 1 (average = 1)
...if 'config int_over_controls 2', these metabolites would be returned in the query.

**Comment by joefutrelle** [comment]
Got it, thanks

**Comment by KujawinskiLaboratory** [comment]
The new exclude_attrs should also act as we talked about exclude_controls above. 

Using only the Tps6 data in the database, if you run the following:
config exclude_controls false
exclude treatment = withoutTps
search 104.1839 74 output.csv

There should be nothing on the list. Instead, right now only the controls are being deleted from the list, leaving behind the metabolites that are in the samples.

**Comment by joefutrelle** [comment]
I believe this is fixed now on the attrs branch.
